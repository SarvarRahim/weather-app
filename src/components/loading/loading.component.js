import React from 'react';
import { hoc } from '@utils/hoc';
import { useLoadingProps } from './loading.props';
// import { LoadingContainer } from './loading.style';

/**
 * <Loading />
 */
const Loading = hoc(useLoadingProps, () => (
  <div className='loader'>
    <div className='loader-logo'>
      <img src='/nt-logo.png' alt="Najot Ta'lim logotype" />
    </div>
    <div className='loader-content'>
      <div className='loader-title' data-testid='loader-title'>
        “Najot Ta’lim”ga xush kelibsiz. <br />
        Sahifa yuklanyapti, uzoq kutib qolmaysiz.
      </div>
      <div className='loader-spinner'></div>
    </div>
  </div>
));

export { Loading };
