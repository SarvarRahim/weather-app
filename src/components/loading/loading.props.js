/**
 * <Loading /> props
 */
export const useLoadingProps = _ => {};
