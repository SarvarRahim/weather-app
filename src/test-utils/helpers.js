import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
