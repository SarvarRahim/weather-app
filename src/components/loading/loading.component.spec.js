import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';
import { Loading } from './loading.component';

test('Should render Loading component', () => {
  const { asFragment } = customRender(<Loading />);

  expect(asFragment()).toMatchSnapshot();
});

test('Loading component should have title and description', () => {
  customRender(<Loading />);
  const loaderTitle = screen.getByTestId('loader-title');

  expect(loaderTitle.innerHTML).toBe(
    `“Najot Ta’lim”ga xush kelibsiz. <br>Sahifa yuklanyapti, uzoq kutib qolmaysiz.`
  );
});
