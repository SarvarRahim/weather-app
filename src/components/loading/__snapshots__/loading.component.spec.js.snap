// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Loading component 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    <div
      class="loader-logo"
    >
      <img
        alt="Najot Ta'lim logotype"
        src="/nt-logo.png"
      />
    </div>
    <div
      class="loader-content"
    >
      <div
        class="loader-title"
        data-testid="loader-title"
      >
        “Najot Ta’lim”ga xush kelibsiz. 
        <br />
        Sahifa yuklanyapti, uzoq kutib qolmaysiz.
      </div>
      <div
        class="loader-spinner"
      />
    </div>
  </div>
</DocumentFragment>
`;
